datanode/src/slippy_util.py

```diff
@@ -0,0 +1,322 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+# logging is our log infrastructure used for this application
+import logging
+import math
+from shapely.geometry import LineString
+from shapely.geometry import Polygon
+
+# Configure the logger
+logging.basicConfig(format='%(levelname)s:%(message)s', level=logging.INFO)
+log = logging.getLogger('InterUSS_DataNode_SlippyUtil')
+
+
+def validate_slippy(z, x, y, raise_error=False):
+  """Validates slippy tile ranges.
+
+  https://en.wikipedia.org/wiki/Tiled_web_map
+  https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames
+
+  Args:
+    z: zoom level in slippy tile format
+    x: x tile number in slippy tile format
+    y: y tile number in slippy tile format
+  Returns:
+    true if valid, false if not
+  """
+  try:
+    z = int(z)
+    x = int(x)
+    y = int(y)
+    if not 0 <= z <= 20:
+      raise ValueError('Zoom must be an integer from 0-20')
+    if not 0 <= x < 2**z:
+      raise ValueError('x must be an integer from 0-%d' % (2**z - 1))
+    if not 0 <= y < 2**z:
+      raise ValueError('y must be an integer from 0-%d' % (2**z - 1))
+    return True
+  except (ValueError, TypeError) as e:
+    log.error('Invalid slippy format for tiles %sz, %s, %s!',
+              str(z), str(x), str(y))
+    log.error('Invalid slippy format error: %s', e)
+    if raise_error:
+      raise e
+    else:
+      return False
+
+
+def convert_csv_to_coordinates(csv):
+  """Converts and validates string of CSV coords into array of lat/lon coords."""
+  if not csv:
+    raise TypeError('CSV of coordinates must be a string in lat,lon[,lat,lon]...format')
+  result = []
+  coords = csv.split(',')
+  if len(coords) % 2 != 0:
+    raise ValueError('CSV of coordinates must in lat,lon pairs')
+  log.debug('Split coordinates to %s and passed early validation...', coords)
+  for a, b in _pairwise(coords):
+    lat = float(a)
+    lon = float(b)
+    if lat >= 90 or lat <= -90 or lon >= 180 or lon <= -180:
+      raise ValueError('CSV of coordinates must have valid lat,lon values')
+    result.append((lat, lon))
+  return result
+
+
+def convert_tile_to_polygon(zoom, xtile, ytile):
+  """Conversion from tile to lat,lon polygon at specific zoom.
+
+  Args:
+      zoom: level of zoom in slippy terms (0-20)
+      xtile: x tile in slippy terms
+      ytile: y tile in slippy terms
+  Raises:
+      TypeError if parameters are not ints
+      ValueError if parameters are not in respective bounds
+  Returns:
+      list of points that make up the tile
```

Comment:
nit: list of (lat, lng) points that make up the tile

Reply:
Done.